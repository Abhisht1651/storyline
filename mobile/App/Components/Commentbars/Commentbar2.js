// @flow

import React, { useState, useEffect } from 'react'
import { View, TextInput, TouchableOpacity } from 'react-native'
import styles, { scaleHeight } from '../Styles/CommentbarStyles/Commentbar2Style'
import Attach from '../../Images/Icons/icon-attach.svg'

type Props = {
  captureText: (arg: string) => void,
  placeholderText: string
}

const Commentbar2 = (props: Props) => {
  const [value, setValue] = useState('')
  const [newStyle, setNewStyle] = useState({ height: 39.87 * scaleHeight })
  const [containerStyle, setContainerStyle] = useState({ height: 90 * scaleHeight })

  const { captureText, placeholderText } = props

  useEffect(() => {
    captureText(value)
  }, [value])

  const changeTextHandler = (text: string) => {
    setValue(text)
  }
  const updateSize = (newHeight) => {
    setNewStyle({
      height: newHeight + scaleHeight * 21.86
    })
    setContainerStyle({
      height: newHeight + scaleHeight * 50.14
    })
  }

  return (
    <View style={[styles.container, containerStyle]} elevation={-1}>
      <TouchableOpacity style={styles.attachIcon}>
        <Attach height="100%" />
      </TouchableOpacity>
      <TouchableOpacity style={[styles.textInput, newStyle]}>
        <TextInput
          placeholder={placeholderText}
          placeholderTextColor="rgba(27, 32, 98, 0.3)"
          selectionColor="#7B46E4"
          multiline
          maxLength={150}
          autoCorrect={false}
          spellCheck={false}
          style={styles.comment}
          onChangeText={changeTextHandler}
          onContentSizeChange={(e) => updateSize(e.nativeEvent.contentSize.height)}
        />
      </TouchableOpacity>
    </View>
  )
}

Commentbar2.defaultProps = {
  placeholderText: 'Comment Something'
}

export default Commentbar2
