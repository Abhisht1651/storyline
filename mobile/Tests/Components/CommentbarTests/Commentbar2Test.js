import { shallow } from 'enzyme'
import React from 'react'
import Attach from '../../../App/Images/Icons/icon-attach.svg'
import Commentbar2 from '../../../App/Components/Commentbars/Commentbar2'

describe('Commentbar2', () => {
  it('should match to snapshot', () => {
    const component = shallow(<Commentbar2 />)
    expect(component).toMatchSnapshot()
  })
  it('should have Attach Icon', () => {
    const component = shallow(<Commentbar2 />)
    expect(component.find(Attach)).toBeTruthy()
  })
})
