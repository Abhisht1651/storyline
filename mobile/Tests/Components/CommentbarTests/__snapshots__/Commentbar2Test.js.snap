// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commentbar2 should match to snapshot 1`] = `
<View
  elevation={-1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E6ECED",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "width": 750,
      },
      Object {
        "height": 147.85714285714286,
      },
    ]
  }
>
  <ForwardRef
    style={
      Object {
        "marginLeft": 51.5,
        "marginRight": 34.5,
      }
    }
  >
    <SvgMock
      height="100%"
    />
  </ForwardRef>
  <ForwardRef
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F7",
          "borderRadius": 19,
          "height": 65.4842857142857,
          "paddingLeft": 15.73,
          "width": 568,
        },
        Object {
          "height": 65.50071428571428,
        },
      ]
    }
  >
    <Component
      allowFontScaling={true}
      autoCorrect={false}
      maxLength={150}
      multiline={true}
      onChangeText={[Function]}
      onContentSizeChange={[Function]}
      placeholder="Comment Something"
      placeholderTextColor="rgba(27, 32, 98, 0.3)"
      rejectResponderTermination={true}
      selectionColor="#7B46E4"
      spellCheck={false}
      style={
        Object {
          "color": "#1B2062",
          "fontFamily": "BananaGrotesk-Light",
          "fontSize": 27,
          "height": "100%",
          "letterSpacing": -0.09,
          "margin": 0,
          "padding": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </ForwardRef>
</View>
`;
